### PR TITLE
Update install-theos for legacy devices

### LINKED ISSUE #797: [Bug]: Install on Fresh iOS 8.4.1 EtasonJB
### What are the steps to reproduce this issue?

On a fresh iOS 8.4.1 install with EtasonJB (don't install any additional packages) attempt the install instructions for Prior to iOS 12.0.

### What happens?

apt-get, head, cut aren't available so this cannot complete.

### What were you expecting to happen?

apt-get, head, and cut available to complete the install.

### Paste any relevant logs, error output, etc.

_No response_

### Additional context

_No response_

### What OS are you seeing the problem on?

iOS

### What OS version does it have installed?

iOS 8.4.1

### What toolchain and version are you using?

N/A

### Which SDK version are you using?

N/A

### What OS is your client device running?

iOS

### What OS version does it have installed?

iOS 8.4.1

### Which device model is it?

iPad 4

### Which jailbreak is it using (if relevant)?

EtasonJB

### PR DESCRIPTION
**Pull request by BJamin99**
What does this implement/fix? Explain your changes.
---------------------------------------------------
add checks for further potentially missing dependencies coreutils and xz-utils.

Does this close any currently open issues?
------------------------------------------
Potentially closes #797 

Any relevant logs, error output, etc?
-------------------------------------
No

Any other comments?
-------------------
No

Where has this been tested?
---------------------------
**Operating System:** 8.4.1

**Platform:** …

**Target Platform:** …

**Toolchain Version:** …

**SDK Version:** …

### CODE REVIEW

File: bin/install-theos [hunk 1]
```diff
@@ -230,6 +230,18 @@ darwin_mobile() {
 		exit 3
 	fi
 
+	# Check for coreutils (not installed by default on some jbs)
```
Comment (L1ghtmann) [inline bin/install-theos:233]: Thanks a ton for looking into this! Did we also want to add apt(-get) here to ensure the later commands proceed?

**Comment by Spoou** [comment]
Good

**Review by Spoou — COMMENTED** [review]
Good

File: bin/install-theos (hunk 1 above)
Comment (L1ghtmann) [reply bin/install-theos:233]: Thanks! And, yeah, good point. That should probably be wrapped in a !$LEGACY check and prompt about the lack of Swift support. Would you be comfortable adding that to this pr? If not, no worries.

**Review by L1ghtmann — APPROVED** [review]
Looks great. Thanks again for bringing this to our attention and for the fix!

**Comment by L1ghtmann** [comment]
Leaving one final note that while moving the iOS Swift install into the non-legacy section prevents attempted installs on pre-64 devices, where it will fail as the toolchain requires arm64*, it also prevents the installer from installing `swift-toolchain` on arm64 devices that are on iOS < 12. 

One way we can adjust this in the future, if desired, is by adding an addition check for `sysctl.cputype` & `sysctl.cpusubtype`, the former of which appears to be `16777228` for arm64* and the latter of which seems to be `1` for 64 and `2`(?) for 64e.

**Pull request merged**